#VRML_SIM R2020b utf8
WorldInfo {
  title "Regbot box"
  basicTimeStep 1
  FPS 25
  physicsDisableLinearThreshold 0.001
  physicsDisableAngularThreshold 0.001
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.5224299886317152 -0.6738114728535055 -0.5225370857932051 4.338968518328391
  position 0.14725851883425747 2.66885322976432 -0.35598905455739016
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 1 0 0 0
  floorSize 5 5
  floorTileSize 15 15
  floorAppearance PBRAppearance {
    baseColor 0.2 0.2 0.2
    baseColorMap ImageTexture {
      url [
        "/home/chr/webots_proj/regbot_3/robocup_2019.png"
      ]
      repeatS FALSE
      repeatT FALSE
      filtering 3
    }
    roughness 1
    metalness 0
  }
  wallThickness 0.05
  wallHeight 0.2
  wallAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "textures/red_brick_wall.jpg"
      ]
      filtering 2
    }
    roughness 1
    metalness 0
  }
  wallColor 1 1 1
  wallTileSize 0.05 0.05
}
WoodenBox {
  translation 0.57634 0.05 0.520356
  size 0.1 0.1 0.1
  mass 1
}
WoodenBox {
  translation 1.20589 0.07 0.79068
  rotation 0 1 0 1.0471976931004259
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 0.2
}
DEF Tape_lines Solid {
  translation 0 0.001 0
  children [
    Transform {
      translation -0.8 0 0.5
      children [
        DEF tape-part-1m Shape {
          appearance DEF tape PBRAppearance {
            baseColor 1 0.8 0.8
            roughness 1
            metalness 0
          }
          geometry Plane {
            size 0.043 1
          }
          castShadows FALSE
        }
      ]
    }
    DEF Arc_90deg Transform {
      translation -0.3 0 0
      children [
        Transform {
          children [
            DEF tape-10cm Shape {
              geometry DEF tape_10cm Plane {
                size 0.1 0.043
              }
            }
          ]
        }
        Transform {
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.086 0 -0.0068
          rotation 0 1 0 0.1571
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.1721 0 -0.0269
          rotation 0 1 0 0.3142
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.2497 0 -0.0599
          rotation 0 1 0 0.4712
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.3233 0 -0.105
          rotation 0 1 0 0.6283
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.3889 0 -0.1611
          rotation 0 1 0 0.7854
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.445 0 -0.2267
          rotation 0 1 0 0.9425
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.49 0 -0.3003
          rotation 0 1 0 1.0996
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.5231 0 -0.38
          rotation 0 1 0 1.2566
          children [
            USE tape-10cm
          ]
        }
        Transform {
          translation 0.5432 0 -0.464
          rotation 0 1 0 1.4137
          children [
            USE tape-10cm
          ]
        }
      ]
    }
    Transform {
      translation -0.8 0 0
      rotation 0 1 0 1.57
      children [
        USE tape-part-1m
      ]
    }
  ]
  name "floor_tape"
  model "aaa"
  description "Tape"
}
DEF BALL Solid {
  translation 2.046191 0.22999999999999998 0.039665
  children [
    DEF BALL_GEOMETRY Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "textures/red_brick_wall.jpg"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Sphere {
        radius 0.03
        subdivision 2
      }
    }
  ]
  description "aaa"
  boundingObject USE BALL_GEOMETRY
  physics Physics {
  }
}
DEF REGBOT_3 Robot {
  translation -1 0.03 0
  rotation 1 0 0 -1.57
  children [
    DEF To_Z_UP Transform {
      translation 0 0 0.019
      rotation 1 0 0 0
      children [
        GPS {
        }
        DEF MAIN_BODY Group {
          children [
            DEF PHYSICS_BOXE_FWD Solid {
              translation 0.05 0 0.03
              name "solid(1)"
              boundingObject Box {
                size 0.14 0.14 0.04
              }
              physics Physics {
                density -1
                mass 0.3
              }
            }
            DEF PHYSICS_BOX_REAR Solid {
              translation -0.03 0 0.08
              rotation 0 1 0 0.7853981610000004
              name "solid(9)"
              boundingObject Box {
                size 0.09 0.16 0.04
              }
              physics Physics {
                density -1
                mass 0.3
              }
            }
            DEF BOTTOM Solid {
              translation 0.055 0 0
              children [
                Shape {
                  appearance DEF PAM PBRAppearance {
                    baseColor 0.9 0.9 0.9
                    roughness 0.5
                    metalness 0
                  }
                  geometry DEF PAM_BOTT Box {
                    size 0.105 0.125 0.003
                  }
                }
              ]
              name "solid(4)"
            }
            DEF PAM_LS_SIDE Solid {
              translation -0.042613198035304736 0 0.055534812083988375
              rotation 0 1 0 0.9162996938995747
              children [
                Shape {
                  appearance USE PAM
                  geometry USE PAM_BOTT
                }
              ]
              name "solid(5)"
            }
            DEF PAM-top Solid {
              translation 0.03133976645910609 0 0.09232050684538894
              rotation 0 1 0 0.5235996938995747
              children [
                Shape {
                  appearance USE PAM
                  geometry DEF PAM_TOP Box {
                    size 0.095 0.125 0.003
                  }
                }
              ]
              name "solid(6)"
            }
            DEF side-left Solid {
              translation 0 0.063 0
              children [
                DEF REGBOT_SIDE Shape {
                  appearance PBRAppearance {
                    baseColor 0.9 0.9 0.9
                    baseColorMap ImageTexture {
                      url [
                        "/home/chr/regbot/WB_proj/regbot_3/protos/textures/dtulogo_256x256.png"
                      ]
                      repeatS FALSE
                      repeatT FALSE
                    }
                    roughness 0.5
                    metalness 0
                  }
                  geometry IndexedFaceSet {
                    coord Coordinate {
                      point [
                        0 0 -0.015
                        0.1 0 -0.015
                        0.11 0 0
                        0.12 0 0.03
                        0.09 0 0.06
                        -0.025 0 0.13
                        -0.07 0 0.11
                        -0.08 0 0.1
                        -0.08 0 0.08
                        0 0.003 -0.015
                        0.1 0.003 -0.015
                        0.11 0.003 0
                        0.12 0.003 0.03
                        0.09 0.003 0.06
                        -0.025 0.003 0.13
                        -0.07 0.003 0.11
                        -0.08 0.003 0.1
                        -0.08 0.003 0.08
                      ]
                    }
                    coordIndex [
                      0, 1, 2, 3, 4, 5, 6, 7, 8, 0, -1, 9, 17, 16, 15, 14, 13, 12, 11, 10, 9, -1, 0, 8, 17, 9, 0, -1, 8, 7, 16, 17, 8, -1, 7, 6, 15, 16, 7, -1, 6, 5, 14, 15, 6, -1, 5, 4, 13, 14, 5, -1, 4, 3, 12, 13, 4, -1, 3, 2, 11, 12, 3, -1, 2, 1, 10, 11, 2, -1, 1, 0, 9, 10, 1, -1
                    ]
                  }
                }
              ]
              name "REGBOT_SIDE"
            }
            DEF Side-right Solid {
              translation 0 -0.066 0
              children [
                USE REGBOT_SIDE
              ]
              name "solid(3)"
            }
          ]
        }
        DEF WHEELS_AND_MOTORS Group {
          children [
            DEF MOTOR_STATIC Solid {
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.3 0.3 0.5
                    roughness 1
                    metalness 0
                  }
                  geometry DEF TWO_MOTORS_STATIC Cylinder {
                    height 0.14
                    radius 0.015
                  }
                }
              ]
              boundingObject USE TWO_MOTORS_STATIC
              physics Physics {
              }
            }
            DEF LEFT_WHEEL HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 1 0
                anchor 0 0.08 0
                dampingConstant 0.001
              }
              device [
                DEF WHEEL_ENCODER PositionSensor {
                  name "POLOLU_D25_ENC_LEFT"
                }
                DEF POLOLU_D25 RotationalMotor {
                  name "POLOLU_D25_LEFT"
                  maxVelocity 100
                }
              ]
              endPoint Solid {
                translation 0 0.08 0
                rotation -0.008227277186992859 -0.9999661274636579 0.00023629520489933751 1.8385883349404006e-06
                children [
                  DEF LEFT_WHEEL_SOLID Solid {
                    children [
                      DEF WHEEL_HUB Shape {
                        appearance PBRAppearance {
                          baseColorMap ImageTexture {
                            url [
                              "textures/wood.jpg"
                            ]
                          }
                          roughness 1
                          metalness 0
                        }
                        geometry Cylinder {
                          height 0.0065
                          radius 0.025
                          subdivision 6
                        }
                      }
                      DEF WHEEL_GEO+APP Shape {
                        appearance PBRAppearance {
                          baseColor 0.2 0.2 0.2
                          roughness 1
                          metalness 0
                        }
                        geometry DEF WHEEL Cylinder {
                          height 0.006
                          radius 0.03
                        }
                      }
                    ]
                    boundingObject USE WHEEL
                    physics Physics {
                      density 600
                    }
                  }
                ]
                name "solid(2)"
                boundingObject USE WHEEL
                physics Physics {
                  density -1
                  mass 0.2
                }
              }
            }
            DEF RIGHT_WHEEL HingeJoint {
              jointParameters HingeJointParameters {
                axis 0 1 0
                anchor 0 -0.08 0
                dampingConstant 0.001
              }
              device [
                PositionSensor {
                  name "POLOLU_D25_ENC_RIGHT"
                }
                RotationalMotor {
                  name "POLOLU_D25_RIGHT"
                  maxVelocity 100
                }
              ]
              endPoint Solid {
                translation 0 -0.08 0
                rotation -3.884728076426398e-06 -0.9999999999924469 1.2109300507237018e-07 1.5619942762111416e-06
                children [
                  USE WHEEL_HUB
                  USE WHEEL_GEO+APP
                ]
                name "solid(7)"
                boundingObject USE WHEEL
                physics Physics {
                  density -1
                  mass 0.2
                }
              }
            }
          ]
        }
        DEF CASTOR_BRANCH Solid {
          translation 0.11 0 0
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.2 0.2 0.3
                roughness 1
                metalness 0
              }
              geometry DEF CASTOR_AXLE Cylinder {
                height 0.02
                radius 0.001
              }
            }
            Transform {
              rotation 1 0 0 1.5708
              children [
                DEF CASTOR_ROTATE HingeJoint {
                  jointParameters HingeJointParameters {
                    axis 0 0 1
                    dampingConstant 0.0002
                  }
                  endPoint Solid {
                    translation -0.008000000000000168 -6.505213034913027e-19 0.02129999999999905
                    rotation 0 0 1 0
                    children [
                      HingeJoint {
                        jointParameters HingeJointParameters {
                          axis 0 1 0
                        }
                        endPoint Solid {
                          children [
                            Shape {
                              appearance PBRAppearance {
                                baseColor 0.3 0.3 0.3
                                roughness 1
                                metalness 0
                              }
                              geometry DEF FRONT_WHEEL Cylinder {
                                height 0.01
                                radius 0.015
                              }
                            }
                          ]
                          boundingObject USE FRONT_WHEEL
                          physics Physics {
                            density -1
                            mass 0.1
                            centerOfMass [
                              0 0 0
                            ]
                            damping Damping {
                            }
                          }
                        }
                      }
                    ]
                    boundingObject DEF FRONT_WHEEL Cylinder {
                      height 0.01
                      radius 0.015
                    }
                    physics Physics {
                    }
                  }
                }
              ]
            }
          ]
          name "solid(8)"
          boundingObject USE CASTOR_AXLE
          physics Physics {
          }
        }
        DEF IMU Transform {
          translation 0.04 -0.03 0.08
          rotation 0 1 0 0.5235987739999998
          children [
            Accelerometer {
              name "imu_acc"
            }
            Gyro {
              name "imu_gyro"
            }
          ]
        }
        DEF IR_SENSORS Group {
          children [
            DistanceSensor {
              translation 0.075980762353278 -0.06500000041525023 0.04
              rotation 0 0 1 5.23598774
              children [
                DEF SHARP_80cm Shape {
                  appearance PBRAppearance {
                    baseColor 0.2 0.2 0.2
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.013 0.01 0.02
                  }
                }
              ]
              name "ds_1_right"
              lookupTable [
                0 0 0
                0.82 0.82 0
              ]
            }
            DistanceSensor {
              translation 0.07500000001384168 0.029999999875424923 0.0713397455226234
              rotation 0.9351131277508578 -0.2505628066116351 -0.25056280300736683 4.645351469128925
              children [
                USE SHARP_80cm
              ]
              name "ds_2_forward"
              lookupTable [
                0 0 0
                0.82 0.82 0
              ]
            }
          ]
        }
        DEF LED_STATUS Transform {
          translation 0 0 0.11699999999999999
          rotation 0 1 0 0.5235987740000014
          children [
            LED {
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Sphere {
                    radius 0.007
                  }
                }
              ]
              name "led1"
              color [
                1 0 0
                0 1 0
                0 0 1
              ]
            }
          ]
        }
        DEF LED_LINESENSOR Transform {
          translation 0 0.02 0.11699999999999999
          rotation 0 1 0 0.5235987740000014
          children [
            LED {
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Sphere {
                    radius 0.007
                  }
                }
              ]
              name "led2"
              color [
                0 1 0
                0 0 1
                1 0 0
              ]
            }
          ]
        }
        SpotLight {
          attenuation 0 0 1
          beamWidth 0.5
          color 1 0 0.8
          cutOffAngle 0.5
          location 0.05 0.05 0
          radius 50
        }
        DEF LS0 DistanceSensor {
          translation 0.035 0.0245 -0.002
          rotation 0 1 0 1.57
          name "ls0"
          lookupTable [
            0 0 0
            0.2 3500 0
          ]
          type "infra-red"
          aperture 45
        }
        DEF LS1 DistanceSensor {
          translation 0.035 0.0175 -0.002
          rotation 0 1 0 1.57
          name "ls1"
          lookupTable [
            0 0 0
            0.2 3500 0
          ]
          type "infra-red"
        }
        DEF LS2 DistanceSensor {
          translation 0.035 0.0105 -0.002
          rotation 0 1 0 1.57
          name "ls2"
          lookupTable [
            0 0 0
            0.2 3500 0
          ]
          type "infra-red"
        }
        DEF LS3 DistanceSensor {
          translation 0.035 0.0035 -0.002
          rotation 0 1 0 1.57
          name "ls3"
          lookupTable [
            0 0 0
            0.2 3500 0
          ]
          type "infra-red"
        }
        DEF LS4 DistanceSensor {
          translation 0.035 -0.0035 -0.002
          rotation 0 1 0 1.57
          name "ls4"
          lookupTable [
            0 0 0
            0.2 3500 0.001
          ]
          type "infra-red"
        }
        DEF LS5 DistanceSensor {
          translation 0.035 -0.0105 -0.002
          rotation 0 1 0 1.57
          name "ls5"
          lookupTable [
            0 0 0
            0.2 3500 0.001
          ]
          type "infra-red"
        }
        DEF LS6 DistanceSensor {
          translation 0.035 -0.0175 -0.002
          rotation 0 1 0 1.57
          name "ls6"
          lookupTable [
            0 0 0
            0.2 3500 0.001
          ]
          type "infra-red"
        }
        DEF LS7 DistanceSensor {
          translation 0.035 -0.0245 -0.002
          rotation 0 1 0 1.57
          name "ls7"
          lookupTable [
            0 0 0
            0.2 3500 0.001
          ]
          type "infra-red"
        }
      ]
    }
  ]
  name "regbot_3"
  model "with castor (HW4)"
  description "First attempt of a REGBOT simulator"
  boundingObject Sphere {
    radius 0.01
  }
  physics Physics {
  }
  controller "regbot_3"
}
